### PR TITLE
Fix test script silent failure

### PR DESCRIPTION
**Pull request by tcharding**
Change the test script to exit with non-zero status code if any command fails. 

The `test.sh` script is silently failing, that means changes causing failures are slipping through our CI pipeline and being merged. 

Resolves: #419 

## Note

Just the last 3 patches, the first 6 are from #420. re-base just shows it works on top of 420, it is going to have to be rebased again when 420 merges.

### CODE REVIEW

**Review by apoelstra — DISMISSED** [review]
ACK e080754181d81426d1fd56dd95c0f597f75a2e8d

Big improvement! Though maybe we shouldn't merge it while WASM is broken.

**Comment by tcharding** [comment]
I noticed the `clang` sanity check is missing in one case and a little wrong in the other. I pushed an additional patch to fix that.

> Though maybe we shouldn't merge it while WASM is broken.

It shouldn't get through CI till then ;)

**Comment by apoelstra** [comment]
utACK 51a79fb2481d4f4d8da8ae0dc971c6338546ec39

**Comment by tcharding** [comment]
Bringing to your attention @apoelstra that this PR removes the pipes instead of using the option you proposed someplace else to make any command in a pipe fail (`set -o pipefail` I think it was).

**Comment by apoelstra** [comment]
Yep, I am aware -- I suggested `pipefail` after this PR was open, actually :). (I'm also not sure if `pipefail` is actually what we want, since these are `&&` connectors rather than pipes..)

In any case, I think it's clearer to have `-e` on and to not use `&&`. Thanks.

File: contrib/test.sh
Comment (thomaseizinger) [inline contrib/test.sh:100]: What is the benefit of explicitly saying exit 0? Isn't that the default?

**Review by thomaseizinger — DISMISSED** [review]
ACK 51a79fb2481d4f4d8da8ae0dc971c6338546ec39

I think this needs a rebase onto https://github.com/rust-bitcoin/rust-secp256k1/pull/420 so we have `clang-9` available for the WASM stuff.

File: contrib/test.sh
Comment (tcharding) [reply contrib/test.sh:100]: The default exit status is the exit status of the last command, because we use `set -e` to exit the script early if any command fails then it is going to be 0. I like to do it explicitly though.

**Comment by tcharding** [comment]
> I think this needs a rebase onto https://github.com/rust-bitcoin/rust-secp256k1/pull/420 so we have clang-9 available for the WASM stuff.

Yeah, I'll rebase once #419, #420, #421 merge.

**Comment by tcharding** [comment]
> I think this needs a rebase onto https://github.com/rust-bitcoin/rust-secp256k1/pull/420 so we have clang-9 available for the WASM stuff.

Yes, now that #421 merged #420 is the only one left :)

**Comment by tcharding** [comment]
Rebased on #420, might be able to merge without me rebasing again, let's see :)

**Comment by apoelstra** [comment]
Looks like both GIthub and my local commit-labelling tool are smart enough to do the right thing, since #420 is merged. (Well, Github still shows "9 commits" and doesn't indicate that they're shared, but it doesn't demand a rebase, so all good.)

**Review by apoelstra — APPROVED** [review]
ACK 97dc0ea9acdc308cc2cadc11e4787d07ec9b2c79

Thank you for doing this!

**Pull request merged**
